src/components/TodoListItem.vue

```diff
@@ -1,34 +1,245 @@
+          </span>
+        </div>
+        <TodoPriorityDropdown
+          v-else
+          v-model="item"
+          @changed-priority="closePriorityChange"
+        />
       </div>
-      <div class="flex items-end justify-between gap-6">
-        <span class="break-words text-2xl font-bold text-gray-500">{{ item.description }}</span>
-        <TodoCheckbox v-model="item" />
+
+      <!-- Description -->
+      <div class="flex max-w-[90%] items-end justify-between gap-6">
+        <span
+          ref="descriptionHTML"
+          class="hidden break-words text-2xl font-bold sm:block"
+          :class="[textColor, priorityDropdownFilter, mobileForceDisplayClass]"
+          :contenteditable="isEditMode && !isPriorityChange"
+          >{{ item.description }}</span
+        >
       </div>
     </div>
+
+    <!-- Save and Delete Buttons -->
+    <div
+      v-if="isEditMode"
+      class="flex items-end gap-2 pb-4 pl-4 pt-4"
+    >
+      <BaseButton
+        :disabled="isPriorityChange"
+        text-color="text-white"
+        background-color="bg-green-500"
+        button-text="Save"
+        :class="priorityDropdownButtonFilter"
+        @click="leaveEditMode"
+      />
+      <BaseButton
+        :disabled="isPriorityChange"
+        text-color="text-black"
+        background-color="bg-gray-300"
+        button-text="Delete"
+        :class="priorityDropdownButtonFilter"
+        @click="openDeleteWindow"
+      />
+    </div>
+
+    <!-- Checkbox -->
+    <div
+      v-if="!isEditMode"
+      class="absolute left-7 top-[35%] sm:left-[91%] sm:top-full sm:-translate-y-12"
+    >
+      <TodoCheckbox v-model="item" />
+    </div>
+
+    <!-- Delete Confirmation Modal -->
+    <Teleport
+      v-if="isDeleteWindow"
+      to="body"
+    >
+      <TodoDeleteModal
+        @close="closeDeleteWindow"
+        @delete="deleteItem"
+      />
+    </Teleport>
   </div>
 </template>
 
 <script setup lang="ts">
-import { computed } from 'vue'
+import { computed, ref } from 'vue'
 import { Todo } from '@/types/Todo'
 import TodoCheckbox from '@/components/TodoCheckbox.vue'
+import BaseButton from '@/components/BaseButton.vue'
+import TodoCalendar from '@/components/TodoCalendar.vue'
+import TodoPriorityDropdown from '@/components/TodoPriorityDropdown.vue'
+import TodoMobilePriorityEditRadio from '@/components/TodoMobilePriorityEditRadio.vue'
+import TodoDeleteModal from '@/components/TodoDeleteModal.vue'
 
 const item = defineModel<Todo>({ required: true })
+const emit = defineEmits<{
+  deleteItem: [id: number]
+}>()
+const isEditMode = ref(false)
+const isPriorityChange = ref(false)
+const isDeleteWindow = ref(false)
 
 const backgroundColorMap = {
   High: 'bg-red-500',
   Medium: 'bg-orange-400',
   Low: 'bg-teal-400'
 }
 
+const titleHTML = ref<HTMLSpanElement | null>()
+const descriptionHTML = ref<HTMLSpanElement | null>()
 const backgroundColor = computed(() => backgroundColorMap[item.value.priority])
+const textColor = computed(() => {
```

Comment:
Please use this computed like this: `isEditMode.value ? 'text-black' : 'text-gray-500'`. 

Also the basic text color is black no?